include/carma_bits/converters.h

```diff
@@ -685,6 +685,31 @@ inline py::array_t<T> to_numpy(armaT* src, int copy = 0) {
     return details::construct_array<T>(data);
 } /* to_numpy */
 
+template <typename armaT, typename T = typename armaT::elem_type, is_Mat_fixed_only<armaT> = 4>
```

Comment:
@yaqwsx Strongly recommend **not** to use `arma::is_Mat_fixed_only<>`.  This is internal Armadillo functionality, which is subject to change or removal without notice.  If this is used, there are no guarantees that it will work with future releases of Armadillo.

Only the [documented API](https://arma.sourceforge.net/docs.html) has stability guarantees.  Extract: "...     any function, class, constant or other code not explicitly described in the public API documentation is considered as part of the underlying internal implementation details, and may be removed or changed without notice. (In other words, don't use internal functionality)."

Reply:
Then I am confused; you already use `is_Mat` and `is_Row` that aren't covered in [documented API](https://arma.sourceforge.net/docs.html).

What is your preferred way to support fixed-size matrices?

Reply:
I only spotted this issue now -- this project (CARMA) is mostly written and maintained by @RUrlus

In any case, this will work for now and is not likely to break within the lifetime of the Armadillo 11.x series.  However, I don't recommend using Armadillo's internal functionality as it may change in the future.  (For example, all of the internal functionality may be renamed and/or moved in Armadillo 12.x).

A possible solution is to clone (reimplement) the functionality of `arma::is_Mat_only` and `arma::is_Mat_fixed_only` directly within CARMA.  That way if `arma::is_Mat_only` etc are ever changed, CARMA's own versions will still stay the same.

Reply:
@conradsnicta You're right, although carma relies on more internal functions e.g. ```arma::access::rw```, these can easily be avoided. I am in the (slow) process of a bit of a rewrite and will fix this.